### DISCUSSION TITLE
Include external/legacy package to manifest

### DISCUSSION THREAD

**Discussion by hwaxman** [webpack/webpack#17114] [Q&A]
Is there anyway to include an external JS package and not have it compiled/chunked but let webpack hash the contents and track in the manifest file?

I have a legacy codebase where I'm trying to remove Gulp and have all the assets managed by webpack. Some of the dependencies (Mootools) are not module compatible and should be loaded globally before the rest.

The legacy output would not use webpack loader, just loaded as is.

I tried the `asset/resource` config but it looks like webpack is trying to compile the code and I'm at a loss for where to go from there.

*Edit to remove misunderstanding.*

**Comment by alexander-akait** [comment]
Where you get this file format?

**Reply by hwaxman** [reply]
Ah.. my mistake; that was from Symfony Encore. I've edited the question to remove that misunderstanding.

**Reply by alexander-akait** [reply]
Sorry, I don't undestand, do you want this https://github.com/webpack-contrib/imports-loader? Also you can use `externals` and just load this code before loading webpack runtime/chunks

**Reply by alexander-akait** [reply]
Sorry, I don't undestand, do you want this https://github.com/webpack-contrib/imports-loader? Also you can use `externals` and just load this code before loading webpack runtime/chunks

**Reply by hwaxman** [reply]
Those will likely be part of the solution.

But what I'm looking for is the asset management aspect: where webpack would copy these externals to the target *as-is*, rename with the content hash, and track in the manifest.

The goal is to eliminate having two build tools, and juggling two manifests; but the older code does not play well with the webpack chunking/loading.

**Reply by alexander-akait** [reply]
You can use `copy-webpack-plugin` in this case, sorry, we don't generate a manifest file (we have an issue about it), so if you don't have it in your manifest file, please open an issue in a plugin repo which you are use
